### PR TITLE
Lawlib legislation page

### PR DESCRIPTION
**Pull request by nickmwangemi**
- Created `LegislationTable/index.vue` to display Legislation data
- Implemented search functionality in component
- Implemented sorting (ascending or descending) by column heading
- Implemented filtering using `SecondaryFacets.vue` 
- Added mobile support. SecondaryFacets are displayed Offcanvas element on mobile view

DEMO:
https://www.loom.com/share/a7de2bde771742c5b044ba7270a98630

### CODE REVIEW

File: lawlibrary/views/legislation.py [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+        return sorted(qs, reverse=True)
+
+    def get_taxonomies(self):
+        taxonomies = []
+        for legislation in self.get_queryset().filter(
+            locality__code=self.kwargs["code"]
+        ):
+            for taxonomy in legislation.taxonomies.all():
+                if taxonomy.topic.name not in taxonomies:
+                    taxonomies.append(taxonomy.topic.name)
```
Comment (ro6ley) [inline lawlibrary/views/legislation.py:120]: You can replace these for loops with `self.get_queryset().filter(locality__code=self.kwargs["code"]).values_list('taxonomies__topic__name', flat=True)`

File: lawlibrary/views/legislation.py [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+            .distinct()
+        )
+        return sorted(qs, reverse=True)
+
+    def get_taxonomies(self):
+        taxonomies = []
+        for legislation in self.get_queryset().filter(locality=None):
+            for taxonomy in legislation.taxonomies.all():
+                if taxonomy.topic.name not in taxonomies:
+                    taxonomies.append(taxonomy.topic.name)
```
Comment (ro6ley) [inline lawlibrary/views/legislation.py:73]: See previous comment on `ProvincialLegislationListView.get_taxonomies`

File: lawlibrary/views/legislation.py [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+    def get_jurisdictions(self):
+        return (
+            self.get_queryset()
+            .order_by("jurisdiction__name")
+            .values_list("jurisdiction__name", flat=True)
+            .distinct()
+        )
+
+    def get_queryset(self):
+        return super().get_queryset()
```
Comment (ro6ley) [inline lawlibrary/views/legislation.py:56]: Can do away with this

File: lawlibrary/views/legislation.py [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+        context["legislation_table"] = LegislationSerializer(qs, many=True).data
+        context["facet_data"] = {
+            "years": self.get_years(),
+            "taxonomies": self.get_taxonomies(),
+        }
+
+        return context
+
+    def get_queryset(self):
+        return super().get_queryset()
```
Comment (ro6ley) [inline lawlibrary/views/legislation.py:101]: This can also fall away

**Review by longhotsummer — APPROVED** [review]

**Pull request merged**